### ISSUE TITLE
[sort-type-union-intersection-members] Sort is formatting-dependent

### ISSUE THREAD

**Issue by Quanyails** [typescript-eslint/typescript-eslint#3639]
I'm in a project that uses Redux, where our actions are defined as a discriminated union keyed by `type`. Some of these actions have long names, causing Prettier to format those union members onto multiple lines. I'd like order this union, but it seems like the ordering changes depending on the formatting.

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/sort-type-union-intersection-members": ["error"]
  }
}
```

```TS
export type Actions1 =
  | { type: "a" }
  | { type: "b" }
  | { type: "c" }
  | { type: "d" };

export type Actions2 =
  | { type: "a" }
  | { type: "b" }
  | { type: "c" }
  | {
      type: "d";
    };
```

**Expected Result**

No changes to the ordering.

**Actual Result**

The `Actions1` type passes the ESLint check, but we receive a linter violation from the `Actions2` object at

```
8:3 - Union type Actions2 members must be sorted.(@typescript-eslint/sort-type-union-intersection-members)
```

If we try applying the autofix, `Actions2` gets sorted into:

```TS
export type Actions2 =
  | {
      type: "d";
    }
  | { type: "a" }
  | { type: "b" }
  | { type: "c" };
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.28.3` |
| `@typescript-eslint/parser`        | `4.24.0` |
| `TypeScript`                       | `4.2.4` |
| `ESLint`                           | `7.17.0` |
| `node`                             | `12.13.1` |

**Comment by bradzacher** [comment]
This is working as designed. The rule is simple in its implementation and it does not interrogate the types - it simply sorts them by their source code text.

Happy to accept a PR if you want to make it smarter!

**Comment by JoshuaKGoldberg** [comment]
Note that we're feature-freezing `sort-type-constituents` and other naming & sorting stylistic rules: #8792. Since this received a PR (#8656) before #8792 was filed, we'll still accept that PR. But if #8656 is closed then we'll close this issue too.

**Comment by JoshuaKGoldberg** [comment]
With #8656 closed, we'll go ahead and close this issue out too. Cheers!
